src/components/tree-detail/watering-dialog.tsx

```diff
@@ -68,6 +70,7 @@ export const WateringDialog: React.FC<WateringDialogProps> = ({
 						disabled={false}
 						onClick={async () => {
 							await waterTree(amount, waterDate);
+							await refreshAdoptedTreesInfo();
```

Comment:
i'd suggest to put this function call rather in the `waterTree` function than below it in the onClick listener